src/chips/sam4l/dma.rs

```diff
@@ -212,6 +216,25 @@ impl DMAChannel {
 
         volatile_store(&mut registers.interrupt_enable, 1 << 1);
     }
+
+    pub fn do_xfer_buf(&self, pid: usize, 
+                       buf: Option<&'static mut [u8]>,
+                       len: usize) {
+        let ptr = buf.unwrap();
+        if len > ptr.len() {
```

Comment:
Ah -- there should be a min() and that is passed to the counter_reload. Don't want to slice because I assume at some point the event will pass the buffer back (we use actual ownership).